src/ast/Event.ts

```diff
@@ -213,6 +213,39 @@ export default class Event extends Node {
   }
 
   typeCheck(map: any): void {
+    // typecheck date
+    if (this.date && !moment(this.date, "MMMM D, YYYY").isValid()) {
```

Comment:
we check to make sure the field exists because in some cases (repeating, all day) they'll be left blank. Since the parser comes first and checks for empty fields, we know if a field is blank at the typechecking stage, it's legit